src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -440,6 +440,7 @@ private void onShardClose(ShardLock lock, boolean ownsShard) {
                     indicesServices.deleteShardStore("delete index", lock, indexSettings);
                 }
             } catch (IOException e) {
+                indicesServices.addPendingDelete(index(), lock.getShardId(), indexSettings);
                 logger.warn("{} failed to delete shard content", e, lock.getShardId());
```

Comment:
I think this can be debug now and we should do the warn once all retires failed, no?